src/GenAPI/Microsoft.DotNet.GenAPI/INamedTypeSymbolExtension.cs

```diff
@@ -159,5 +160,56 @@ public static bool IsBoundGenericType(this INamedTypeSymbol namedType)
 
             return true;
         }
+
+        // Synthesize an internal default constructor for the type with implicit default constructor and the base type without.
+        public static IEnumerable<SyntaxNode> TryGetInternalDefaultConstructor(this INamedTypeSymbol namedType, ISymbolFilter symbolFilter)
+        {
+            if (!namedType.Constructors.Where(symbolFilter.Include).Any() && namedType.BaseType != null)
```

Comment:
Switch the condition checks as the `namedType.BaseType != null` is O(1) while the former condition is at least O(n).